libpimeval/src/pimParamsPerf.h

```diff
@@ -23,25 +23,39 @@ class pimParamsPerf
   pimParamsPerf(pimParamsDram* paramsDram);
   ~pimParamsPerf() {}
 
-  double getMsRuntimeForBytesTransfer(uint64_t numBytes) const;
-  double getMsRuntimeForFunc1(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForRedSum(PimCmdEnum cmdType, const pimObjInfo& obj, unsigned numPass) const;
-  double getMsRuntimeForBroadcast(PimCmdEnum cmdType, const pimObjInfo& obj) const;
-  double getMsRuntimeForRotate(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  class perfEnergy
+  {
+    public:
+      double m_msRuntime;
+      double m_mjEnergy;
+  };
+
+  perfEnergy getPerfEnergyForBytesTransfer(uint64_t numBytes) const;
+  perfEnergy getPerfEnergyForFunc1(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForFunc2(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForRedSum(PimCmdEnum cmdType, const pimObjInfo& obj, unsigned numPass) const;
+  perfEnergy getPerfEnergyForBroadcast(PimCmdEnum cmdType, const pimObjInfo& obj) const;
+  perfEnergy getPerfEnergyForRotate(PimCmdEnum cmdType, const pimObjInfo& obj) const;
```

Comment:
can we have separate functions for runtime and energy?

Reply:
Based on previous discussion, it's okay to compute both runtime and energy in same function for now since there are lots of shared code. Some refactoring can be done later.